### PR TITLE
DM-40441: first batch of deprecations from RFC-949

### PR DESCRIPTION
**Pull request by TallJimbo**
## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/pipe/base/graph/quantumNode.py [hunk 1]
```diff
@@ -101,6 +102,19 @@ class QuantumNode:
     creation.
     """
 
+    @property
+    def task_node(self) -> TaskNode:
```
Comment (timj) [inline python/lsst/pipe/base/graph/quantumNode.py:106]: Is this property something that can easily be added to a test?

File: python/lsst/pipe/base/pipeTools.py [hunk 2]
```diff
@@ -44,13 +48,31 @@
 if TYPE_CHECKING:
     from .taskFactory import TaskFactory
 
+# TODO: remove this module on DM-40443.
+warnings.warn(
```
Comment (timj) [inline python/lsst/pipe/base/pipeTools.py:52]: Doesn't look like anything in lsst_distrib or ci is using this code. GitHub shows no-one using it either (other than forks of pipe_base) so removing this file completely seems entirely possible.

File: python/lsst/pipe/base/pipeline.py [hunk 3]
```diff
@@ -601,6 +596,17 @@ def findSubsetsWithLabel(self, label: str) -> set[str]:
                 results.add(subset.label)
         return results
 
+    @property
+    def task_labels(self) -> Set[str]:
```
Comment (timj) [inline python/lsst/pipe/base/pipeline.py:600]: Add a test?

File: python/lsst/pipe/base/pipeline_graph/_pipeline_graph.py [hunk 4]
```diff
@@ -260,6 +261,47 @@ def __deepcopy__(self, memo: dict) -> PipelineGraph:
         # that mutable state is copied.
         return self.copy()
 
+    def diff_tasks(self, other: PipelineGraph) -> list[str]:
```
Comment (timj) [inline python/lsst/pipe/base/pipeline_graph/_pipeline_graph.py:264]: Is it possible to test this in pipe_base?

File: python/lsst/pipe/base/tests/simpleQGraph.py [hunk 5]
```diff
@@ -458,37 +505,36 @@ def makeSimpleQGraph(
...
+        "Instantiating QuantumGraphBuilder, "
+        "skip_existing_in=%s, input_collections=%r, output_run=%r, where=%r, bind=%s.",
+        skipExistingIn,
         butler.collections,
         run,
         userQuery,
         bind,
     )
+    if not run:
+        assert butler.run is not None, "Butler must have run defined"
```
Comment (timj) [inline python/lsst/pipe/base/tests/simpleQGraph.py:521]:
```suggestion
        assert butler.run is not None, "Butler must have run defined if run not passed in as a parameter"
```

**Review by timj — APPROVED** [review]

File: python/lsst/pipe/base/pipeTools.py (hunk 2 above)
Comment (TallJimbo) [reply python/lsst/pipe/base/pipeTools.py:52]: Given that I've already got the deprecation in, I figure it'll still be easier for me to leave it as is and drop it on DM-40443.

**Pull request merged**